File: simple-stack-example-sharedelement-fragment/src/main/java/com/example/fragmenttransitions/BaseFragment.java
package com.example.fragmenttransitions;

import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;

/**
 * Created by Owner on 2017. 08. 08..
 */

public class BaseFragment
        extends Fragment {
    @NonNull
    public final <T extends BaseKey> T getKey() {
        T key = getArguments() != null ? getArguments().getParcelable("KEY") : null;
        if(key == null) {
            throw new NullPointerException("Key should not be null");
        }
        return key;
    }
}

File: simple-stack-example-basic-kotlin/src/main/java/com/zhuinden/simplestackexamplekotlin/BaseFragment.kt
package com.zhuinden.simplestackexamplekotlin

import android.support.v4.app.Fragment

/**
 * Created by Owner on 2017.11.13.
 */

open class BaseFragment : Fragment() {
    fun <T : BaseKey> getKey(): T = requireArguments.getParcelable<T>("KEY")
}
